src/main/java/com/google/api/codegen/config/FieldConfig.java

```diff
@@ -212,10 +212,14 @@ public FieldConfig withResourceNameConfig(ResourceNameConfig resourceNameConfig)
   public boolean requiresParamTransformation() {
     return getResourceNameConfig() != null
         && getMessageResourceNameConfig() != null
-        && getMessageResourceNameConfig().getResourceNameType() != ResourceNameType.ANY
         && !getResourceNameConfig().equals(getMessageResourceNameConfig());
   }
 
+  public boolean requiresAnyParamTransformation() {
```

Comment:
This is an older problem (the generated test cases were broken even without the recent resourcename changes, see the `ListStrings` method test case). So, even though the need to convert both is currently present only in test cases, it can potentially be relevant to non-test cases. Specifically when we talk about resource names there are 3 resource name types:

Regular resource name: `BookName` (for example)
Oneof resource name: `BookNameOneof`
AnyResource name: essentially resolves to `UntypedResourceName`.

All this 3 may have been initialized from exactly same string representation. Their `toString()` methods will return exactly same value, from the server side point of view they are equal. But as resourcename class instances all 3 will be distinct (**all** of the 3 possible pairs will be `x.equals(y) == false`). 

The fact that: `ResourceName instance (x) -> Server side representation (y) -> Resource Name instance (z)` conversion chain may (and often does) end up in` x.equals(z) == false` feels disturbing, and maybe the we should either change `equals` implementation of for those classes or at least introduce some other method in ResourceName interface like `equalsToResource()` or something like that.